app/javascript/controllers/loading_controller.js

```diff
@@ -0,0 +1,9 @@
+import { Controller } from "@hotwired/stimulus";
+
+export default class extends Controller {
+  show() {
+    const loadingOverlays = document.querySelectorAll(".loading-overlay");
```

Comment:
Suggest we use a stimulus target for this. Makes it explicit both in stimulus and html markup what is hooked up to what (one of my favourite things about Stimulus 😊).